runtime/kernel/src/sys/linux/stringmgr.cpp

```diff
@@ -312,31 +312,31 @@ void str_FreeString(HSTRING hString)
 	}
 }
 
-
 bool str_CompareStrings(HSTRING hString1, HSTRING hString2)
 {
 	std::string String1( ((StringWrapper*)hString1)->m_Bytes);
 	std::string String2( ((StringWrapper*)hString2)->m_Bytes);
 
-	if( !String1.empty() || !String2.empty() )
+	if (String1.empty() || String2.empty())
 		return false;
 
-    return (String1 == String2);
+	return (String1 == String2);
 }
 
-
 bool str_CompareStringsUpper(HSTRING hString1, HSTRING hString2)
 {
 	ci_string String1( ((StringWrapper*)hString1)->m_Bytes);
 	ci_string String2( ((StringWrapper*)hString2)->m_Bytes);
 
-	if( !String1.empty() || !String2.empty() )
+	std::transform(String1.begin(), String1.end(), String1.begin(), ::toupper);
+	std::transform(String2.begin(), String2.end(), String2.begin(), ::toupper);
+
+	if (String1.empty() || String2.empty())
```

Comment:
Could we have this done prior to the 2 transforms? 

Exit quick if there's no work.

Reply:
Done.